src/main/java/ru/testtask/service/BackgroundJobService.java

```diff
@@ -0,0 +1,114 @@
+package ru.testtask.service;
+
+
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Service;
+import ru.testtask.dto.BackgroundJobDTO;
+import ru.testtask.model.BackgroundJob;
+import ru.testtask.model.BackgroundJobMongo;
+import ru.testtask.model.BackgroundJobStatus;
+import ru.testtask.repo.BackgroundJobRepo;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class BackgroundJobService {
+    @Autowired
+    private BackgroundJobRepo backgroundJobRepo;
+
+    private final ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
```

Comment:
Will this task scheduler work without @EnableScheduling annotation?

Reply:
It is better to use "fixed thread pool executor" to set limit of threads to execute in parallel